### PR TITLE
[Property API] Fallback to route entry cluster name

### PR DESCRIPTION
**Pull request by bianpengyuan**
Signed-off-by: Pengyuan Bian <bianpengyuan@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/common/wasm/wasm.cc [hunk 1]
```diff
@@ -1194,7 +1194,11 @@ WasmResult Context::getProperty(absl::string_view path, std::string* result) {
         value = CelValue::CreateMessage(plugin_->listener_metadata_, &arena);
         break;
       case PropertyToken::CLUSTER_NAME:
-        value = CelValue::CreateString(&info->upstreamHost()->cluster().name());
+        if (!info->upstreamHost()->cluster().name().empty()) {
+          value = CelValue::CreateString(&info->upstreamHost()->cluster().name());
+        } else {
+          value = CelValue::CreateString(&info->routeEntry()->clusterName());
```
Comment (kyessenov) [inline source/extensions/common/wasm/wasm.cc]: I think routeEntry can be null? In any case, many of these pointers can be null, so we should harden it at some point.

Comment (bianpengyuan) [reply source/extensions/common/wasm/wasm.cc]: thanks! Done.

**Comment by bianpengyuan** [comment]
@kyessenov @jplevyak @PiotrSikora  ptal thanks!

**Comment by kyessenov** [comment]
LGTM but I can't approve.

**Pull request merged**